tests/v1/test_build_request.py

```diff
@@ -106,3 +106,67 @@ def test_signature():
 
     assert params['api_signature'] == hashlib.sha1(
         '{}{}'.format(base_str, SECRET).encode('utf-8')).hexdigest()
+
+def test_signature_with_array_values():
+
+    KEY = 'api_key'
+    SECRET = 'api_secret'
+    PATH = '/test/resource/show'
+
+    request_params = {
+        'a': 1,
+        'b': 'two',
+        'c3': 'Param 3',
+        u'❄': u'⛄',
+        't1': True,
+        'n0': None,
+        'test_array1': [1, 2, 3, 4],
+        'test_array2': ["test item1", "test item2"],
+    }
+
+    jwp_client = jwplatform.v1.Client(KEY, SECRET)
+
+    url, params = jwp_client._build_request(PATH, request_params)
+
+    assert url == 'https://api.jwplatform.com/v1{}'.format(PATH)
+    assert 'api_nonce' in params
+    assert 'api_timestamp' in params
+    assert 'api_key' in params
+    assert 'api_format' in params
+    assert 'api_kit' in params
+    assert 'api_signature' in params
+
+    request_params['api_nonce'] = params['api_nonce']
+    request_params['api_timestamp'] = params['api_timestamp']
+    request_params['api_key'] = params['api_key']
+    request_params['api_format'] = params['api_format']
+    request_params['api_kit'] = params['api_kit']
+
+    # The logic before allowing array in the query string
+    # The array in the query string will be like "key=[val1,val2]"
+    base_str = '&'.join(['{}={}'.format(
+        quote(str(key).encode('utf-8'), safe='~'),
+        quote(str(value).encode('utf-8'), safe='~')
+    ) for key, value in sorted(request_params.items())])
+
+    assert params['api_signature'] != hashlib.sha1(
+        '{}{}'.format(base_str, SECRET).encode('utf-8')).hexdigest()
+
+
+    # The array in the query string will be like "key=val1&key=val2"
+    params_for_sbs = list()
```

Comment:
The output here should probably be hard-coded instead of duplicating the code that generates it. Cam's suggestion to parameterize this would help with doing so.